nnsmith/abstract/op.py

```diff
@@ -361,6 +361,44 @@ def name(cls) -> str:
         return cls.__name__.split(".")[-1]
 
 
+@mark_materialize("core")
+class ConcreteOp(AbsOpBase):
+    class empty:
+        """placeholder"""
```

Comment:
Second thought: "x is ConcreteOp.Placeholder" means `x` is a class. We don't want to make it a class as class means reusable and initializable! For example, `enum` should be a better choice. For simplicity, we can move it out of `ConcreteOp`.

It can be as simple as:

```python
from enum import Enum, auto

class PH(Enum): # fx style naming
    TENSOR = auto()

# check with x == PH.TENSOR
```

Reply:
I see but I don't feel `inspect.Parameter.empty` is a good design IMO. Here we basically need a "marker" which looks like a constant/macro/etc. Using "class-in-class" seems to be a bit sophisticated IMO and is not very eye-catching.

Reply:
Using `obj is class` as a predicate also seems incommon...

Reply:
😂 Okay let's ship it as it is now. I will take the refactor later.